config/schemas/config.yaml

```
@@ -126,6 +411,654 @@ properties:
+              annotationsRiskLevel:
+                type: string
+                description: See [reference](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#annotations-risk-level)
+                enum:
+                  - Critical
+                  - High
+                  - Medium
+                  - Low
+            type: object
+          enablepublishService:
+            default: false
+            type: boolean
+          enableAnnotationValidations:
+            description: |-
+              Enable validation of Nginx annotations on Ingress resources.
+              This is kept false by default due to the maturity of the feature,
+              e.g. lack of documentation and disruptive bugs.
+            default: false
+            type: boolean
+          extraArgs: true
+          nodeSelector:
+            $ref: '#/$defs/nodeSelector'
+          resources:
+            $ref: '#/$defs/resources'
+          service:
+            additionalProperties: false
+            properties:
+              annotations:
+                additionalProperties:
+                  type: string
+                type: object
+              clusterIP:
+                format: ipv4
+                type: string
+              enabled:
+                type: boolean
+              loadBalancerSourceRanges:
+                items:
+                  type: string
+                type: array
+              ipFamilyPolicy:
+                type: string
+                description: |-
+                  Represents the dual-stack-ness requested or required by this Service. When
+                  utilizing an internal loadbalancer service (ie MetalLB), set this field to
+                  "RequireDualStack" if you want both IPv4 and IPv6 connectivity. The
+                  ipFamilies and clusterIPs fields depend on the value of this field.
+
+                  See [reference](https://kubernetes.io/docs/concepts/services-networking/dual-stack/)
+                enum:
+                  - SingleStack
+                  - PreferDualStack
+                  - RequireDualStack
+                default: SingleStack
+              ipFamilies:
+                type: array
+                description: |-
+                  List of IP families (e.g. IPv4, IPv6) assigned to the service. Default is IPv4 only. When utilizing an internal loadbalancer service (ie MetalLB),
+                  IPv6 would also need to be included in order for the ingress service to allocate an address in that family.
+                uniqueItems: true
+                items:
+                  type: string
+                  enum:
+                    - IPv4
+                    - IPv6
+                default:
+                  - IPv4
+              nodePorts:
+                additionalProperties: false
+                properties:
+                  http:
+                    default: 30080
+                    type: integer
```

I tested this out and I got some errors due to invalid type for `type: integer`, as we define e.g. ports as strings in our config. I think it is preferable if we use type string over type integer [as mentioned in the Helm docs as to avoid casting issues](https://helm.sh/docs/chart_best_practices/values/#make-types-clear), unless required by an upstream helm chart.
```suggestion
                    type: string
```

---

I would expect unsigned 16 bit integers to survive most casting. Isn't it better to handle that coercion in the templating layer, keeping port numbers consistently as integers in the config?

Won't numbers as strings get cast back into numbers when passed trough the yaml templating and parsing unless they're explicitly quoted?

Based on grepping the READMEs in `helmfile.d/upstream` I gather that many port settings can be set to the empty string. Also about half port settings seem to be `int` (of those easily greppable, probably missed a ton). 

```
$ find helmfile.d/ -name README.md -exec grep -w port {} +|grep -w string|wc -l
31
$ find helmfile.d/ -name README.md -exec grep -w port {} +|grep -w int|wc -l
28
```

Both could be supported with
```suggestion
                    type:
                      - string
                      - integer
```

---

I would be fine with supporting both string and integer in this case.

---

Do we use symbolic names anywhere? I.e. how in some contexts `"https"` is a valid port number. If not then 0-5 digits could be enforced by regex.